### ISSUE TITLE
Distinct is not pushed down to connectors

### ISSUE THREAD

**Issue by fx19880617** [prestodb/presto#14809]
Hi,

I'm trying to upgrade to recent presto, and below query are not getting the correct plan pushed down:
```
SELECT distinct A FROM myTable LIMIT 10
```

From the PlanOptimizer, the maxSubPlan we are getting is the TableScanNode now.
![image](https://user-images.githubusercontent.com/1202120/86959603-f27be780-c112-11ea-9b01-8bf6d2634adf.png)

And below is the query plan generated:
```
presto:default> explain select distinct teamid from baseballstats limit 10;

--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 - Output[teamid] => [teamid:varchar]
     - Project[] => [teamid:varchar]
         - DistinctLimit[10][$hashvalue] => [teamid:varchar, $hashvalue:bigint]
             - LocalExchange[SINGLE] () => [teamid:varchar, $hashvalue:bigint]
                 - RemoteStreamingExchange[GATHER] => [teamid:varchar, $hashvalue_3:bigint]
                     - DistinctLimitPartial[10][$hashvalue_4] => [teamid:varchar, $hashvalue_4:bigint]
                         - ScanProject[table = TableHandle {connectorId='pinot', connectorHandle='PinotTableHandle{connectorId=pinot, schemaName=default, tableName=baseballStats, isQueryShort=Optional[false], expectedColumnHandles=Optio
                                 Estimates: {rows: ? (?), cpu: ?, memory: 0.00, network: 0.00}/{rows: ? (?), cpu: ?, memory: 0.00, network: 0.00}
                                 $hashvalue_4 := combine_hash(BIGINT 0, COALESCE($operator$hash_code(teamid), BIGINT 0))
                                 teamid := PinotColumnHandle{columnName=teamID, dataType=varchar, type=REGULAR}
```

**Comment by fx19880617** [comment]
cc @zhenxiao @agrawaldevesh

**Comment by mbasmanova** [comment]
@fx19880617 

>  query failed as it's not getting the correct plan pushed down

What is the failure? Would you paste the stack trace?

**Comment by mbasmanova** [comment]
CC: @rongrong @kaikalur

**Comment by mbasmanova** [comment]
CC: @highker

**Comment by fx19880617** [comment]
> @fx19880617
> 
> > query failed as it's not getting the correct plan pushed down
> 
> What is the failure? Would you paste the stack trace?

It's the pushdown logic not as expected, basically underlying full table scan will bringdown the connected db.

**Comment by mbasmanova** [comment]
I wonder if this is because com.facebook.presto.sql.planner.plan.DistinctLimitNode is not listed in com.facebook.presto.sql.planner.optimizations.ApplyConnectorOptimization#CONNECTOR_ACCESSIBLE_PLAN_NODES

Could you see what the plan looks like when com.facebook.presto.spi.ConnectorPlanOptimizer#optimize is called? Which nodes are there? (The output of EXPLAIN is not enough because it shows the final plan, not the plan at the time of calling ConnectorPlanOptimizer#optimize.)

**Comment by kaikalur** [comment]
Does it work without the LIMIT? I think there is a different pushdown for DistinctLimit vs Distinct.

**Comment by fx19880617** [comment]
Looks like it goes to `DistinctLimitNode` for query `select distinct teamid from baseballstats limit 10;`
![image](https://user-images.githubusercontent.com/1202120/87145356-0bdd7a80-c25e-11ea-9a6f-7fb7295fa5dc.png)

@kaikalur You are right, by removing the LIMIT, it works as expected with an AggregationNode has only groupingSets and no aggregations.
This query: `select distinct teamid from baseballstats;`

![image](https://user-images.githubusercontent.com/1202120/87145512-49420800-c25e-11ea-8d25-349a2c0a2548.png)

**Comment by fx19880617** [comment]
@mbasmanova I'm thinking of adding `DistinctLimitNode` to `CONNECTOR_ACCESSIBLE_PLAN_NODES` wdyt?

**Comment by mbasmanova** [comment]
> @mbasmanova I'm thinking of adding `DistinctLimitNode` to `CONNECTOR_ACCESSIBLE_PLAN_NODES` wdyt?

Git it a try. CC: @sachdevs

**Comment by fx19880617** [comment]
@mbasmanova 
Could you help review this: https://github.com/prestodb/presto/pull/14831

**Comment by mbasmanova** [comment]
@fx19880617 I see that James already reviewed and merged #14831. Will there be a follow-up PR? If not, let's close this issue.

**Comment by fx19880617** [comment]
> @fx19880617 I see that James already reviewed and merged #14831. Will there be a follow-up PR? If not, let's close this issue.

Not more changes for now, I'm also feeling the SortNode is also pushdown-able, but not relevant to this issue.
I'll make connector changes! Many thanks for your advice and thanks to @highker for reviewing it !
